modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-shipment-impl/src/main/java/com/liferay/headless/commerce/admin/shipment/internal/resource/v1_0/ShipmentResourceImpl.java

```diff
@@ -0,0 +1,297 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.commerce.admin.shipment.internal.resource.v1_0;
+
+import com.liferay.commerce.constants.CommerceShipmentConstants;
+import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.model.CommerceShipment;
+import com.liferay.commerce.service.CommerceAddressService;
+import com.liferay.commerce.service.CommerceCountryService;
+import com.liferay.commerce.service.CommerceOrderService;
+import com.liferay.commerce.service.CommerceRegionService;
+import com.liferay.commerce.service.CommerceShipmentService;
+import com.liferay.headless.commerce.admin.shipment.dto.v1_0.Shipment;
+import com.liferay.headless.commerce.admin.shipment.dto.v1_0.ShippingAddress;
+import com.liferay.headless.commerce.admin.shipment.internal.dto.v1_0.converter.ShipmentDTOConverter;
+import com.liferay.headless.commerce.admin.shipment.internal.util.v1_0.ShippingAddressUtil;
+import com.liferay.headless.commerce.admin.shipment.resource.v1_0.ShipmentResource;
+import com.liferay.headless.commerce.core.util.ServiceContextHelper;
+import com.liferay.petra.function.UnsafeConsumer;
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.search.filter.Filter;
+import com.liferay.portal.kernel.util.CalendarFactoryUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterRegistry;
+import com.liferay.portal.vulcan.dto.converter.DefaultDTOConverterContext;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+import com.liferay.portal.vulcan.util.SearchUtil;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Map;
+
+import javax.validation.constraints.NotNull;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+/**
+ * @author Andrea Sbarra
+ */
+@Component(
+	enabled = false,
+	properties = "OSGI-INF/liferay/rest/v1_0/shipment.properties",
+	scope = ServiceScope.PROTOTYPE, service = ShipmentResource.class
+)
+public class ShipmentResourceImpl extends BaseShipmentResourceImpl {
+
+	@Override
+	public void deleteShipment(Long shipmentId) throws Exception {
+		_commerceShipmentService.deleteCommerceShipment(
+			shipmentId, Boolean.FALSE);
+	}
+
+	@Override
+	public Shipment getShipment(@NotNull Long shipmentId) throws Exception {
```

Comment:
@andrea-ale-sbarra @riccardo-alberti  No need for `@NotNull` since its already handled in the extended base impl.
```suggestion
	public Shipment getShipment(Long shipmentId) throws Exception {
```